core/domain/rte_component_registry_test.py

```diff
@@ -87,94 +91,85 @@ def _listdir_omit_ignored(self, directory):
             names = [name for name in names if not name.endswith(suffix)]
         return names
 
-    def test_allowed_rich_text_components_and_counts(self):
-        """Do sanity checks on the ALLOWED_RTE_EXTENSIONS dict in feconf.py."""
-        self.assertEqual(
-            len(rte_component_registry.Registry.get_all_rte_components()),
-            len(feconf.ALLOWED_RTE_EXTENSIONS))
-
-        for (component_name, component_definition) in (
-                feconf.ALLOWED_RTE_EXTENSIONS.iteritems()):
-            contents = os.listdir(
-                os.path.join(os.getcwd(), component_definition['dir']))
-            self.assertIn('%s.py' % component_name, contents)
-
     def test_image_thumbnails_for_rte_components(self):
         """Test the thumbnails for the RTE component icons."""
-        for (cpt_name, cpt_spec) in feconf.ALLOWED_RTE_EXTENSIONS.iteritems():
-            image_filepath = os.path.join(
-                os.getcwd(), cpt_spec['dir'], '%s.png' % cpt_name)
-
-            with open(image_filepath, 'rb') as f:
+        rte_components = (
+            rte_component_registry.Registry.get_all_rte_components())
+        for (component_name, component_specs) in rte_components.iteritems():
+            generated_image_filepath = os.path.join(
+                os.getcwd(), feconf.RTE_EXTENSIONS_DIR,
+                component_name, '%s.png' % component_name)
+            relative_icon_data_url = component_specs['icon_data_url'][1:]
+            defined_image_filepath = os.path.join(
+                os.getcwd(), feconf.EXTENSIONS_DIR_PREFIX,
+                'extensions', relative_icon_data_url)
+            self.assertEqual(generated_image_filepath, defined_image_filepath)
+
+            with open(generated_image_filepath, 'rb') as f:
                 img_data = f.read()
                 width, height = struct.unpack('>LL', img_data[16:24])
                 self.assertEqual(int(width), RTE_THUMBNAIL_WIDTH_PX)
                 self.assertEqual(int(height), RTE_THUMBNAIL_HEIGHT_PX)
 
-    def test_default_rte_components_are_valid(self):
+    def test_rte_components_are_valid(self):
         """Test that the default RTE components are valid."""
 
-        for component_id in feconf.ALLOWED_RTE_EXTENSIONS:
+        rte_components = (
+            rte_component_registry.Registry.get_all_rte_components())
+
+        for (component_name, component_specs) in rte_components.iteritems():
             # Check that the component id is valid.
-            self.assertTrue(self._is_camel_cased(component_id))
+            self.assertTrue(self._is_camel_cased(component_name))
```

Comment:
Why don't we keep it as component_id? ID implies uniqueness.

Perhaps we should rename backend_name to backend_id as well?

Reply:
Yes, it should be id.

For the _backend_name_ I was trying to be consistent with _frontend_name_.

Reply:
OK. I'm fine with changing both to id, if you like, but up to you.

Reply:
Done.